src/lib/dnssd/Advertiser.h

```diff
@@ -95,20 +95,20 @@ class BaseAdvertisingParams
     const chip::ByteSpan GetMac() const { return chip::ByteSpan(mMacStorage, mMacLength); }
 
     // Common Flags
-    Derived & SetLocalMRPConfig(const Optional<ReliableMessageProtocolConfig> & config)
+    Derived & SetLocalMRPConfig(const std::optional<ReliableMessageProtocolConfig> & config)
     {
         mLocalMRPConfig = config;
         return *reinterpret_cast<Derived *>(this);
     }
-    const Optional<ReliableMessageProtocolConfig> & GetLocalMRPConfig() const { return mLocalMRPConfig; }
+    const std::optional<ReliableMessageProtocolConfig> & GetLocalMRPConfig() const { return mLocalMRPConfig; }
 
     // NOTE: The SetTcpSupported API is deprecated and not compliant with 1.3. T flag should not be set.
-    Derived & SetTcpSupported(Optional<bool> tcpSupported)
+    Derived & SetTcpSupported(std::optional<bool> tcpSupported)
     {
         mTcpSupported = tcpSupported;
         return *reinterpret_cast<Derived *>(this);
     }
-    Optional<bool> GetTcpSupported() const { return mTcpSupported; }
+    const std::optional<bool> & GetTcpSupported() const { return mTcpSupported; }
```

Comment:
Why are we changing this, and the vendor id, etc, to return references?

I guess it's OK, but I was just talking about whether the _callers_ should force copies, not whether the _callees_ should be changed to not copy.  I guess this is fine, probably....

Reply:
Reverted. I thought it odd for us to do `auto &x = foo.GetTcpSupported()` when the `GetTcpSupported()` returns by value rather than reference since then it is somewhat unclear why we even ask for a reference to a temporary. I.e. I assumed this is what you asked for.

If you did not ask for it, I am fine to do minimal code change instead.